google/cloud/forseti/services/client.py

```diff
@@ -724,9 +742,14 @@ def __init__(self, endpoint=DEFAULT_ENDPOINT, ping=False):
         Raises:
             Exception: gRPC connected but services not registered
         """
+        interceptors = create_interceptors(endpoint)
         self.gigabyte = 1024 ** 3
-        self.channel = grpc.insecure_channel(endpoint, options=[
-            ('grpc.max_receive_message_length', self.gigabyte)])
+        self.channel = grpc.intercept_channel(
```

Comment:
What does `grpc.intercept_channel()` do? I looked at `_interceptor.py` and it's not clear what it does.

Reply:
It intercepts a channel through a set of interceptors. It returns a channel that intercepts each invocation via the provided interceptors.